c7n/resources/emr.py

```diff
@@ -12,27 +12,137 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from c7n.manager import resources
+from c7n.actions import ActionRegistry, BaseAction
 from c7n.query import QueryResourceManager
+from c7n import utils
+from c7n.utils import (
+    local_session, type_schema, get_account_id,
+    get_retry, chunks, generate_arn, snapshot_identifier)
 
+actions = ActionRegistry('emr.actions')
 
-@resources.register('emr-cluster')
+
+@resources.register('emr')
 class EMRCluster(QueryResourceManager):
 
+    action_registry = actions
+
+    def __init__(self, ctx, data):
+        super(EMRCluster, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(self.data.get('query', []))
+
     class Meta(object):
         service = 'emr'
-        type = 'emr-cluster'
+        type = 'emr'
         enum_spec = ('list_clusters', 'Clusters', None)
         name = 'Name'
         id = 'Id'
         dimension = 'ClusterId'
 
     resource_type = Meta
-        
+
+    def resources(self, query=None):
+        q = self.consolidate_query_filter()
+        if q is not None:
+            query = query or {}
+            for i in xrange(len(q)):
```

Comment:
just use range instead of xrange, py3 compatible, xrange is only useful in py2 for very large collections.

also for this given usage, i think you just want the enumerate builtin